src/Pim/Bundle/CatalogBundle/spec/Elasticsearch/Filter/IdentifierFilterSpec.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+namespace spec\Pim\Bundle\CatalogBundle\Elasticsearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+use PhpSpec\ObjectBehavior;
+use Pim\Bundle\CatalogBundle\Elasticsearch\Filter\AbstractFieldFilter;
+use Pim\Bundle\CatalogBundle\Elasticsearch\Filter\IdentifierFilter;
+use Pim\Bundle\CatalogBundle\Elasticsearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Exception\InvalidOperatorException;
+use Pim\Component\Catalog\Query\Filter\FieldFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+use Pim\Component\Catalog\Repository\AttributeRepositoryInterface;
+
+class IdentifierFilterSpec extends ObjectBehavior
+{
+    function let(AttributeRepositoryInterface $attributeRepository)
+    {
+        $this->beConstructedWith(
+            $attributeRepository,
+            ['identifier'],
+            [
+                'STARTS WITH',
+                'CONTAINS',
+                'DOES NOT CONTAIN',
+                '=',
+                '!=',
+                'IN LIST',
+                'NOT IN LIST'
+            ]
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(IdentifierFilter::class);
+    }
+
+    function it_is_a_field_filter()
+    {
+        $this->shouldImplement(FieldFilterInterface::class);
+        $this->shouldBeAnInstanceOf(AbstractFieldFilter::class);
```

Comment:
Sorry "extends the Abstract class", in fact we could remove all our check on "$this->shouldBeAnInstanceOf" because it does not express anything about the contract of the interface ;-)